### PR TITLE
Fix/less proposals

### PR DESCRIPTION
**Pull request by dvilelaf**
## Proposed changes

* Retrieve less snapshot proposals, focusing on the ones that expire first
* Drop snapshot proposals at the end of the period since they are retrieved every period anyway, and having too many produces the registration payload too big issue.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

n/a

### CODE REVIEW

File: packages/valory/skills/proposal_collector_abci/rounds.py [hunk 1]
```diff
@@ -267,7 +270,11 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
                 SynchronizedData, self.synchronized_data
             ).snapshot_proposals
             snapshot_proposals.extend(payload["snapshot_proposals"])
-            print(f"Adding {len(payload['snapshot_proposals'])}")
+            finished = (
+                payload["finished"]
+                or len(snapshot_proposals) >= SNAPSHOT_PROPOSAL_TOTAL_LIMIT
+            )
+
```
Comment (dvilelaf) [inline packages/valory/skills/proposal_collector_abci/rounds.py:277]: Even if there are still proposals to retrieve, the first 200 are more than enough for one period.

File: packages/valory/skills/proposal_collector_abci/snapshot.py [hunk 2]
```diff
@@ -31,8 +31,8 @@
     where: {
       state: "active"
     },
-    orderBy: "created",
-    orderDirection: desc
+    orderBy: "end",
+    orderDirection: asc
```
Comment (dvilelaf) [inline packages/valory/skills/proposal_collector_abci/snapshot.py:35]: Get proposals sorted by expiration date

File: packages/valory/skills/proposal_voter_abci/rounds.py [hunk 3]
```diff
@@ -184,6 +184,9 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
                     synchronized_data_class=SynchronizedData,
                     **{
                         get_name(SynchronizedData.proposals): payload["proposals"],
+                        get_name(
+                            SynchronizedData.snapshot_proposals
+                        ): [],  # clear snapshot proposals to avoid too big registration payloads
```
Comment (dvilelaf) [inline packages/valory/skills/proposal_voter_abci/rounds.py:189]: Drop snapshot proposals at the end of the period. We are retrieving them from scratch anyway on the next period, and this way we avoid too big registration payloads.

**Comment by dvilelaf** [comment]
The test failures on windows are due to the custom eth_account folder that was included  last week. Until we can have eth_accounts >= 0.6.1 on the framework (flashbots need to support it first) this was the only solution.

**Review by 0xArdi — APPROVED** [review]

**Review by Adamantios — APPROVED** [review]

**Pull request merged**